### ISSUE TITLE
selector-no-type rule doesn't exclude the nesting selector (&)

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#499]
Ref: https://github.com/stylelint/stylelint/issues/489#issuecomment-155576063

> What about nesting and [type selection](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/rules.md#selector)? This is currently considered an error if `selector-no-type` is turned on and you are nesting per the spec:

```css
.foo {
    color: orange;

    & [class*=bar] {
        color: blue
    }
}
```

> **Edit:** in this instance `postcss-selector-parser` considers the `&` a tag.

Pinging @ben-eb. Is the _nesting selector_ a selector you'd consider parsing for? If so, I'll create an issue in `postcss-selector-parser`. Otherwise, we'll handle it at our end.

**Comment by ben-eb** [comment]
@jeddy3 Yeah, well the `&` had no special meaning before; shows how fast web development moves. :smile:

I'd accept a PR for it, my free time is at a premium right now. Remember this will also be a breaking change so I don't think it will be done soon (there are more breaking changes to be done too, some of which update the API to match PostCSS 5 iterators), so whilst a patch would be welcome I feel it would be a while before we could release a 2.0. However if people would like to help out with that then that'd be awesome!

So I think the easiest solution for now is handling it at your end. But that's not to say that we couldn't support that type of selector for 2.0.

**Comment by jeddy3** [comment]
@ben-eb Thanks for the quick response!

> Yeah, well the & had no special meaning before; shows how fast web development moves.

Tell me about it :)

> Remember this will also be a breaking change... so whilst a patch would be welcome I feel it would be a while before we could release a 2.0. 

Yep, makes sense.

> So I think the easiest solution for now is handling it at your end. But that's not to say that we couldn't support that type of selector for 2.0.

Cool. We'll do that. I'll added a note to your `2.0` issue too.

@davidtheclark Is this a breaking change at our end too (and should I added this to `3.0.0`), or should we mark this up as a bug in our check as the change won't produce any new warnings?

**Comment by davidtheclark** [comment]
I would consider this a patch, not a breaking change or a new feature.

**Comment by jeddy3** [comment]
Agreed.

**Comment by jeddy3** [comment]
Released as `2.3.1`
